harness/tests/experiment/fixtures/estimator_xor_model.py

```diff
@@ -150,3 +151,45 @@ def build_validation_spec(self) -> tf.estimator.EvalSpec:
                 shuffle=False,
             )
         )
+
+
+class CustomCallback(tf.estimator.SessionRunHook):
+    def __init__(self, file_path: str) -> None:
+        self._file_path = file_path
+        self._idx = 0
+
+    def before_run(self, run_context: Any) -> None:
+        self._idx += 1
+
+    def after_run(self, run_context: Any, run_values: Any) -> None:
+        with open(self._file_path, "w") as fp:
+            fp.write(f"{self._idx}")
+            fp.flush()
```

Comment:
non-blocking: i think the `flush()` is redundant since the file closes once the `with` block is exited.

Reply:
removed